packages/amplify-provider-awscloudformation/src/aws-utils/aws-s3.js

```diff
@@ -67,27 +73,61 @@ class S3 {
       }
     });
   }
-
-  deleteAllObjects(bucketName) {
+  getAllObjectKeys(bucketName, continuationToken = null) {
     return new Promise((resolve, reject) => {
       this.s3
-        .listObjects({ Bucket: bucketName })
+        .listObjectsV2({ Bucket: bucketName, ContinuationToken: continuationToken })
         .promise()
         .then((result, lerr) => {
```

Comment:
any reason to not use async / await here? IMO this callback is a little difficult to reason through, especially with the nested promise callback